### PR TITLE
Implement Addressable v1beta1 to use URL.

### PR DESCRIPTION
**Pull request by mattmoor**
Fixes: https://github.com/knative/pkg/issues/395

/hold

### CODE REVIEW

File: apis/duck/v1beta1/addressable_types.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Status AddressStatus `json:"status"`
+}
+
+// AddressStatus shows how we expect folks to embed Addressable in
+// their Status field.
+type AddressStatus struct {
+	Address *Addressable `json:"address,omitempty"`
```
Comment (vagababov) [inline apis/duck/v1beta1/addressable_types.go]: Not inlined?

Comment (mattmoor) [reply apis/duck/v1beta1/addressable_types.go]: This is how it works in `v1alpha1`, which we aren't changing.

**Comment by mattmoor** [comment]
/hold cancel

No real opposition on the issue, so removing the hold\ here

**Pull request merged**